### ISSUE TITLE
Add operations failing for addresses and phoneNumbers

### ISSUE THREAD

**Issue by srisaiswaroop** [thomaspoignant/scim-patch#187]
**Describe the bug**
When scimUser does not have type:work in the addresses array and we do **Add** operation on  "path": "addresses[type eq \"work\"].country",  Patch is failing due to the below error thrown in addOrReplaceObjectAttribute
```
if (patch.op === 'add')
            throw new scimErrors_1.InvalidScimPatchOp('Invalid patch query.');
```

When i comment the above error line, it works perfectly fine. Another solution is to pre populate the address and phoneNumbers  like below (this works for only cases where nothing is present in these fields)
```
 scimUser.addresses = scimUser.addresses || [{ type: 'work' }];
  scimUser.phoneNumbers = scimUser.phoneNumbers || [{ type: 'mobile' }];
```

**To Reproduce**
You can try this on runkit as well. 
```
const {scimPatch} = require("scim-patch")

const scimUser = {"userName":"toffer@yahoo.com","emails":[{"primary":true,"type":"work","value":"scott@hotmail.com"}],"schemas":["urn:ietf:params:scim:schemas:core:2.0:User"],"id":"b83fb889"};

const operations = [{"op":"add","path":"addresses[type eq \"work\"].country","value":"country11"}]
const patched = scimPatch(scimUser, operations)
```

**Expected behavior**
Expecting it to work without throwing the error and add type work if its not already present.  

**Additional context**
Its one of the Microsoft Azure AD test cases that we received when we submitted our API for user provisioning gallery app.

**Comment by thomaspoignant** [comment]
Hi @srisaiswaroop thank you for reporting this issue.
I will review the RFC and the code to validate this issue and get back to you as soon as possible.

**Comment by srisaiswaroop** [comment]
I am using the latest 0.5.7 which was released yesterday.

**Comment by thomaspoignant** [comment]
@srisaiswaroop I was able to reproduce the problem, and we had inconsistency depending if you were using `Add` or `add` as your SCIM operation that's the reason why it was not spotted in the test.

I have added a new test with `add` and fix the issue in the PR #188.

I will deploy a new version today fixing this issue.

**Comment by thomaspoignant** [comment]
The version `v0.5.8` is [available](https://npmjs.org/package/scim-patch).
